src/phase0/state_transition/epoch_processing.rs

```diff
@@ -154,8 +156,63 @@ pub fn process_registry_updates<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    // Process activation eligibility and ejections
+    let current_epoch = get_current_epoch(state, context);
+
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+
+        if is_active_validator(validator, current_epoch)
+            && validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context);
+        }
+    }
+
+    // Queue validators eligible for activation and not yet dequeued for activation
+    let _activation_queue = state
```

Comment:
Here's where step 2 begins & issues begin -- and the type on `_activation_queue` is `()` with the current code. If I remove the `sort_by`, the type is a `Vec<(usize, &Validator)>`, which is what I'd expect. So, I'm thinking it's an issue with my handling of `sort_by` but am not entirely sure here.

Reply:
yeah, i think this is just a chaining issue...

you are calling `collect` to materialize the `Vec` and then immediately calling `sort_by` which mutates the Vec in-place.

but you are capturing the result of this (just the `()` as the sort mutates the vec in place and simply returns `()` here) as `activation_queue` so everything seems as expected.


if i'm understanding correctly, you just need to say `let mut activation_queue = state./*... elided ...*/.collect();` and then on the next line `activation_queue.sort_by(...)`

Reply:
Ah, yeah, that was straightforward...have to have the `activation_queue` created to be able to sort it.